shadows/framework/src/main/java/org/robolectric/shadows/ShadowLegacyAssetManager.java

```diff
@@ -277,7 +272,9 @@ protected void init(boolean isSystem) {
   }
 
   protected ResourceTable getResourceTable() {
-    return resourceTable;
+    return isSystem
```

Comment:
Out of curiosity, why does lazily loading the resource table fix this issue? Why are the resource tables not available in the constructor?

Reply:
Not entirely sure why lazy loading fixes this issue..

Reply:
https://github.com/robolectric/robolectric/pull/6918#issuecomment-989047499 https://github.com/robolectric/robolectric/pull/6918#pullrequestreview-826767998  https://github.com/robolectric/robolectric/pull/6918#issuecomment-989053673 @hoisie those comments maybe provide some explanation.

Reply:
This could have been potentially caused by the refactoring of [AndroidTestEnvronment](https://github.com/robolectric/robolectric/blob/master/robolectric/src/main/java/org/robolectric/android/internal/AndroidTestEnvironment.java) to support `@LazyApplication`. I'm find with this solution for now, maybe at some point we can reorder some of the logic in AndroiddTestEnvironment to potentially fix the root cause.